src/uct/gaudi/base/gaudi_topo.c

```diff
@@ -0,0 +1,1929 @@
+/**
+* Copyright (C) Intel Corporation, 2025. ALL RIGHTS RESERVED.
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "gaudi_topo.h"
+
+#include <ucs/config/global_opts.h>
+#include <ucs/debug/log.h>
+#include <ucs/sys/string.h>
+#include <ucs/sys/sys.h>
+#include <ucs/type/spinlock.h>
+#include <ucs/type/status.h>
+
+#include <ctype.h>
+#include <dirent.h>
+#include <errno.h>
+#include <float.h>
+#include <math.h>
+#include <pthread.h>
+#include <sched.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+
+#ifdef HAVE_GAUDI_TOPO_API
+
+#define COMPARE(a, b)                      ((a) < (b) ? -1 : (a) > (b) ? 1 : 0)
+#define UCT_GAUDI_TOPO_ACCEL_PATH          "/sys/class/accel/"
+#define UCT_GAUDI_TOPO_INFINIBAND_PORT_FMT "/sys/class/infiniband/%s/ports/1/"
+#define UCT_GAUDI_TOPO_VENDOR_ID           0x1da3 /* Habana Labs Vendor id */
+#define UCT_GAUDI_TOPO_MELLANOX_VENDOR_ID  0x15b3
```

Comment:
it is used a lot in ib_device.c, maybe some general constant is worth  introducing